.github/workflows/kafka_demo_test.yml

```diff
@@ -0,0 +1,138 @@
+name: Kafka Demo Test 
+
+on:
+  pull_request:
+    branches: 
+      - main
+  workflow_call:
+    inputs:
+      sidecar-registry:
+        description: "The registry to get the image from"
+        required: true
+        default: "mcr.microsoft.com"
+        type: string
+      key-release-image:
+        description: "The image of the SKR sidecar to use"
+        required: true
+        default: "aci/skr:2.7"
+        type: string
+      consumer-image: 
+        description: "Consumer Image"
+        required: true
+        type: string
+      producer-image: 
+        description: "Producer Image"
+        required: true
+        type: string
+      kubernetes-version:
+        description: "Kubernetes version to use for AKS cluster creation"
+        required: true
+        default: "1.28.3"
+        type: string
+      skr-kafka-demo-topic:
+        description: "Kafka topic"
+        required: true
+        default: "kafka-demo-topic"
+        type: string  
+  workflow_dispatch:
+    inputs:
+      sidecar-registry:
+        description: "The registry to get the image from"
+        required: true
+        type: choice
+        default: mcr.microsoft.com
+        options: 
+          - mcr.microsoft.com
+          - confidentialsidecars.azurecr.io
+      key-release-image:
+        description: "The image of the SKR sidecar to use"
+        required: true
+        default: "aci/skr:2.7"
+        type: string
+      consumer-image: 
+        description: "Consumer Image"
+        default: acc/samples/kafka/consumer:1.0
+        required: true
+        type: string
+      producer-image: 
+        description: "Producer Image"
+        default: acc/samples/kafka/producer:1.0
+        required: true
+        type: string
+      kubernetes-version:
+        description: "Kubernetes version to use for AKS cluster creation"
+        required: true
+        default: "1.28.3"
+        type: string
+  merge_group:
+    branches:
+      - main
+
+jobs:
+  prepare:
+    name: Extract Environment Variables
+    runs-on: ubuntu-latest
+    outputs:
+      sidecar-registry: ${{ steps.extract-envs.outputs.sidecar-registry }}
+      key-release-image: ${{ steps.extract-envs.outputs.key-release-image }}
+      consumer-image: ${{ steps.extract-envs.outputs.consumer-image }}
+      producer-image: ${{ steps.extract-envs.outputs.producer-image }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Extract Envs 
+        id: extract-envs
+        run: | 
+          echo "Extracting Environment Variables"
+          echo "Triggering event is ${{ github.event_name }}"
+          if [[ "${{ github.event_name }}" == "pull_request" ]]; then
+            echo "sidecar-registry=mcr.microsoft.com" >> $GITHUB_OUTPUT
+            echo "key-release-image=aci/skr:2.7" >> $GITHUB_OUTPUT
+            echo "consumer-image=acc/samples/kafka/consumer:1.0" >> $GITHUB_OUTPUT
```

Comment:
You don't need to do this. The default value should be picked up from the inputs, if none is selected (as in the case of the pull request event).

Reply:
No, workflow inputs are not available for the pull_request event. Workflow inputs are only available when a workflow is triggered manually using the workflow_dispatch event. I thought they would be available as defaults as well but apparently, they are not, I end up getting empty values in the pipeline.